apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/logging/core/WriterFactory.java

```diff
@@ -25,23 +25,37 @@
 import org.apache.skywalking.apm.agent.core.conf.SnifferConfigInitializer;
 import org.apache.skywalking.apm.util.StringUtil;
 
+/**
+ * @author Alan Lau
+ */
 public class WriterFactory {
+
+    private static volatile IWriter WRITER;
```

Comment:
This variable is not going to be changed in concurrency case, right? Why need `volatile`?

Reply:
As logger.info may be called by many threads, so we give volatile may reduce call the check method.  @wu-sheng

Reply:
Most application has thread pool, such as Tomcat. Even cache mechanism makes the check method invoked more than once, it is harmless. But if you add `volatile `, you will force every single access to real memory rather than cache. I am seeing this as an unexpected cost.

Reply:
ok, i've checked the false sharing doc. You're right. i will remove this.